### PR TITLE
Added tests and fix n+1 problems

### PR DESCRIPTION
**Pull request by alphayowakarindi**
At this juncture, [Mark ](https://github.com/markotuya0), [Pierre](https://github.com/Piercel2022) and I implemented the following:
- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.

-  Use Capybara to write integration tests for each view in the project.
  
- User index page:
  - I can see the username of all other users.
  - I can see the profile picture for each user.
  - I can see the number of posts each user has written.
  - When I click on a user, I am redirected to that user's show page.
  
- user show page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see the user's bio.
  - I can see the user's first 3 posts.
  - I can see a button that lets me view all of a user's posts.
  - When I click a user's post, it redirects me to that post's show page.
  - When I click to see all posts, it redirects me to the user's post's index page.

- User post index page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see a post's title.
  - I can see some of the post's body.
  - I can see the first comments on a post.
  - I can see how many comments a post has.
  - I can see how many likes a post has.
  - I can see a section for pagination if there are more posts than fit on the view.
  - When I click on a post, it redirects me to that post's show page.
  
  - I can see the post's title.
  - I can see who wrote the post.
  - I can see how many comments it has.
  - I can see how many likes it has.
  - I can see the post body.
  - I can see the username of each commentor.
  - I can see the comment each commentor left.

### CODE REVIEW

**Review by gemmen29 — APPROVED** [review]
Hi @alphayowakarindi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/posts/show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  end
+  it 'renders who wrote the post ' do
+    expect(page.html).to include(@user.name)
+  end
+  it 'renders number of comments ' do
+    expect(page).to have_content(1)
+  end
+  it 'renders number of likes ' do
+    expect(page).to have_content(0)
+  end
```
Comment (mikethreels) [inline spec/views/posts/show_spec.rb:22]: - [ ] for the user's post-show page you should write a test that asserts you can see the post's title as well.

**Review by mikethreels — COMMENTED** [review]
Hi Pierre, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that was already approved and merged. please create a separate branch for each review request

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by alphayowakarindi** [comment]
Hi @gemmen29, thank you so much for taking the time to leave this excellent review. Cheers!

**Review by mukhammadrizooff — COMMENTED** [review]
Hi Mark Otuya, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request
You should implement the [these requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md) in all repositories in your `pair-programming` group. That means that each group member needs to open a `new pull` request in `their repo`. But you are using your partner's repo, please fix this issue🙏

image 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request merged**